packages/dhis2/src/Adaptor.js

```diff
@@ -497,25 +503,29 @@ export function update(
  *   trackedEntity:['F8yKM85NbxW'],
  * });
  */
-export function get(resourceType, query, options = {}, callback = false) {
+export function get(resourceType, query, options = {}, callback = s => s) {
   return state => {
     console.log('Preparing get operation...');
 
     const [resolvedResourceType, resolvedQuery, resolvedOptions] =
       expandReferences(state, resourceType, query, options);
 
-    const { params, requestConfig } = resolvedOptions;
+    const { params, requestConfig, asBase64 = false } = resolvedOptions;
     const { configuration } = state;
 
     return request(configuration, {
       method: 'get',
       url: generateUrl(configuration, resolvedOptions, resolvedResourceType),
-      params: { ...resolvedQuery, ...params },
-      responseType: 'json',
+      params: {
+        ...resolvedQuery,
+        ...params,
+      },
+      responseType: asBase64 ? 'arraybuffer' : 'json',
       ...requestConfig,
     }).then(result => {
       console.log(`Retrieved ${resolvedResourceType}`);
-      return handleResponse(result, state, callback);
+      const response = asBase64 ? { data: encode(result.data) } : result;
```

Comment:
This sent me on a bit of a rabbit hole. I think it will be really healthy to rebsae dhis2 on unidic and make it much more standardised